File: 1128-remove-all-adjacent-duplicates-in-string/remove-all-adjacent-duplicates-in-string.ts
function removeDuplicates(s: string): string {
    const stack: string[] = []

    for (const char of s) {
        if (stack.length > 0 && stack[stack.length - 1] === char) {
            stack.pop()
        } else {
            stack.push(char)
        }
    }
    return stack.join('')
};

File: 1128-remove-all-adjacent-duplicates-in-string/remove-all-adjacent-duplicates-in-string.py
class Solution:
    def removeDuplicates(self, s: str) -> str:
        stack = []

        for char in s:
            if len(stack) > 0 and stack[len(stack) - 1] == char:
                stack.pop()
            else:
                stack.append(char)

        return ''.join(stack)